mvcc/kvstore.go

```diff
@@ -379,6 +380,19 @@ func (s *store) restore() error {
 	tx := s.b.BatchTx()
 	tx.Lock()
 
+	v := UnsafeDetectSchemaVersion(s.lg, tx)
+	if !v.Equal(version.V3_4) && !v.Equal(version.V3_5) {
+		if s.lg != nil {
+			s.lg.Panic("unsupported storage version",
+				zap.String("storage-version", v.String()))
+		} else {
+			plog.Panicf("unsupported storage version: %s\n", v.String())
+		}
+	}
+	if v.Equal(version.V3_5) {
+		downgradeMetaBucket(s.lg, tx)
```

Comment:
Suggest to gate this by the option `--next-cluster-version-compatible` as well.

Reply:
the problem with gating this is it might break the upgrade process. 
If the 3.4 cluster is started without `--next-cluster-version-compatible`, when some node is upgraded to 3.5, and some 3.4 node is falling behind. When the 3.4 node receives a snapshot from 3.5 node, we do want to do `downgradeMetaBucket` to avoid any data problems when that node upgrade to 3.5. So this step is needed regardless of `--next-cluster-version-compatible`

Reply:
I understood your point.

- This is a generic problem. I would suggest to fix it in main firstly, then 3.5 and 3.4.
- I think eventually we do need to gate this under a flag just in case it has any regression.

It can be a followup.

Reply:
added flag check.